pipeline/config.go

```diff
@@ -38,6 +38,12 @@ func (c Config) BuildOperators(bc operator.BuildContext) ([]operator.Operator, e
 		}
 		operators = append(operators, op...)
 	}
+
+	if defaultOperator != nil {
```

Comment:
We should also check whether the last operator can output. Otherwise, appending the default operator will have no effect other than inserting an orphaned node into the graph.

```suggestion
	if defaultOperator != nil && operators[len(operators)-1].CanOutput() {
```

Reply:
Added